File: src/test/java/LongestCommonPrefixTest.java
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class LongestCommonPrefixTest {

    @Test
    void test() {
        LongestCommonPrefix longestCommonPrefix = new LongestCommonPrefix();
        assertEquals("fl", longestCommonPrefix.longestCommonPrefix(new String[]{"flower", "flow", "flight"}));
        assertEquals("", longestCommonPrefix.longestCommonPrefix(new String[]{"dog", "racecar", "car"}));
        assertEquals("", longestCommonPrefix.longestCommonPrefix(new String[]{}));
    }
}

File: src/test/kotlin/array/LongestCommonPrefixTest.kt
package array

import org.assertj.core.api.Assertions
import org.junit.jupiter.api.Test

class LongestCommonPrefixTest {
    @Test
    fun `find longest common prefix`() {
        Assertions.assertThat(LongestCommonPrefix().longestCommonPrefix(arrayOf("flower", "flow", "flight"))).isEqualTo("fl")
        Assertions.assertThat(LongestCommonPrefix().longestCommonPrefix(arrayOf("dog","racecar","car"))).isEqualTo("")
        Assertions.assertThat(LongestCommonPrefix().longestCommonPrefix(arrayOf())).isEqualTo("")
        Assertions.assertThat(LongestCommonPrefix().longestCommonPrefix(arrayOf("flower", "", "flight"))).isEqualTo("")
    }
}